main.go

```diff
@@ -0,0 +1,240 @@
+		if m.Type == "image" && m.Action == "push" {
+			go addToImageList(m.ID, url)
+		}
+	}
+	return nil
+}
+
+type image struct {
+	ID   string   `json:"id"`
+	Name string   `json:"name"`
+	Tags []string `json:"tags"`
+}
+
+func addToImageList(name string, url string) {
+	images, err := getRepoDigests(name)
+	if err != nil {
+		logger.Print(err)
+		return
+	}
+
+	for _, i := range images {
+		go func(i image) {
+			attempt := 0
+			for {
+				if attempt == 30 {
+					logger.Printf("error submitting %s image: max retries reached", name)
+					return
+				}
+
+				if attempt != 0 {
+					<-time.After(time.Second * 3)
+				}
+				attempt++
+
+				data, err := json.Marshal(i)
+				if err != nil {
+					logger.Printf("error submitting %q image: %v", i.Name, err)
+					return
+				}
+
+				resp, err := httpPut(url, "application/json", bytes.NewReader(data))
+				if err != nil {
+					logger.Printf("error submitting %q image: %v", i.Name, err)
+					continue
+				}
+				defer func() {
+					io.Copy(ioutil.Discard, resp.Body)
+					resp.Body.Close()
+				}()
+
+				if resp.StatusCode == http.StatusOK {
+					logger.Printf("submitted %+v", i)
+					return
+				}
+
+				if resp.StatusCode == http.StatusInternalServerError {
+					logger.Printf("error submitting %q image: got http status code %d from imagelist", name, resp.StatusCode)
+					continue
+				} else {
+					logger.Printf("error submitting %q image: bad request, status %d from imagelist", name, resp.StatusCode)
+					return
+				}
+			}
+		}(i)
+	}
+}
+
+func getRepoDigests(name string) ([]image, error) {
+	var images []image
+
+	// Could use a global client, but docker client does not implement
+	// auto-reconnect and I don't want to implement this myself.
+	c, err := client.NewEnvClient()
+	if err != nil {
+		return images, err
+	}
+	defer c.Close()
+
+	if !tagRegexp.MatchString(name) {
+		return images, fmt.Errorf("unable to find image %q without a tag", name)
+	}
+
+	ctx := context.Background()
+	imageInspect, _, err := c.ImageInspectWithRaw(ctx, name)
+	if err != nil {
+		return images, err
+	}
+
+	dm := mapRepoDigestsToTags(name, imageInspect)
+	if len(dm) == 0 {
+		return images, fmt.Errorf("unable to find repo digests for %q image", name)
+	}
+
+	for k, v := range mapRepoDigestsToTags(name, imageInspect) {
+		images = append(images, image{k, tagRegexp.ReplaceAllString(name, ""), v})
+	}
+
+	return images, nil
+}
+
+// mapRepoDigestsToTags finds RepoDigests that match image name and returns a
+// map of repoDigest to tags.
+func mapRepoDigestsToTags(name string, image types.ImageInspect) map[string][]string {
+	m := make(map[string][]string)
+	if name == "" {
+		return m
+	}
+
+	// trim a tag from image name if exists
+	name = tagRegexp.ReplaceAllString(name, "")
+
+	tags := []string{}
+	for _, entry := range image.RepoTags {
+		if strings.HasPrefix(entry, name) {
+			matches := tagRegexp.FindStringSubmatch(entry)
+			if len(matches) == 2 {
+				tags = append(tags, matches[1])
+			}
```

Comment:
Is it worth doing some debug logging for cases where this doesn't match? So we can more easily identify edge cases.